comp110-worksheet-6/OxoBoard.cs

```diff
@@ -8,42 +8,89 @@ namespace comp110_worksheet_6
 
-		// Return the contents of the specified square.
-		public Mark GetSquare(int x, int y)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        public OxoBoard()
+        {
+            tictacktoe = new Mark[3, 3]; // initialize that field in the constructor
+        }
 
-		// If the specified square is currently empty, fill it with mark and return true.
-		// If the square is not empty, leave it as-is and return False.
-		public bool SetSquare(int x, int y, Mark mark)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // Return the contents of the specified square.
+        /*
+        public Mark GetSquare(int x, int y)
+        {
+            if (tictacktoe[x, y] == Mark.O) // comparing int and enum - ERROR
+            {
+                return Mark.O;
+            }
+            else if (tictacktoe[x, y] == Mark.X) // comparing int and enum - ERROR
+            {
+                return Mark.X;
+            }
+            else
+            {
+                return Mark.None;
+            }
+        }
+        */
 
-		// If there are still empty squares on the board, return false.
-		// If there are no empty squares, return true.
-		public bool IsBoardFull()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // you can rewrite this function simply like this
+        public Mark GetSquare(int x, int y)
+        {
+            return (Mark)tictacktoe[x, y]; // with an explicit cast
+        }
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
-		public Mark GetWinner()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            tictacktoe[x, y] = mark; // with an explicit cast
+
+            return false;
+        }
+
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+        public bool IsBoardFull()
+                {
+                    for (int y = 0; y < 2; y++)
+                    {
+                        for (int x = 0; x < 2; x++)
+                        {
+                            if (tictacktoe[x, y] == Mark.None)
+                            {
+                                return false;
+                            }
+                        }
+                    }
+
+            return true;
+                }
+
+                // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+                // Otherwise, return Mark.None.
+                public Mark GetWinner()
+                {
+                    Mark GetWinner(Mark player)
```

Comment:
You're defining a nested function (a function within a function) here -- I'm guessing this isn't what you intended?